### PR TITLE
Support Clipping Route Viewer Routes Inside Flex Zones

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR adds a new config item making use of the new feature added in https://github.com/opentripplanner/otp-ui/pull/325.

Tests will fail until the otp-ui PR is merged.

### CODE REVIEW

File: example-config.yml [hunk 1]
```diff
@@ -405,3 +405,7 @@ dateTime:
 #   # departure and it is not entirely excluded from display
 #   # (defaults to 4 days/345600s if unspecified).
 #   timeRange: 345600
+# routeViewer:
+#   # Whether to render routes within flex zones of a route's patterns. If set to true,
+#   # routes will not be rendered within flex zones.
+#   clipRouteRenderToPatternStops: true
```
Comment (binh-dam-ibigroup) [inline example-config.yml:411]: The config parameter name is not very descriptive. Maybe `hideRouteShapesWithinFlexZones`?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I recommend renaming that new config parameter. (https://github.com/opentripplanner/otp-react-redux/pull/519/files#r775587336)

File: example-config.yml (hunk 1 above)
Comment (miles-grant-ibigroup) [reply example-config.yml]: Good call! Renamed in 04daa4b7000472f9a5b547d37936429ef0824c3b

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Needs package `route-viewer-overlay` 1.2.0.

**Comment by binh-dam-ibigroup** [comment]
@miles-grant-ibigroup I think an updated `yarn.lock` is still missing, could you add it please?

**Review by binh-dam-ibigroup — APPROVED** [review]

**Review by daniel-heppner-ibigroup — APPROVED** [review]

**Pull request merged**